src/main/java/org/opentripplanner/model/vehicle_position/RealtimeVehiclePosition.java

```diff
@@ -0,0 +1,70 @@
+package org.opentripplanner.model.vehicle_position;
+
+import java.time.Instant;
+import org.opentripplanner.model.StopLocation;
+
+/**
+ * Internal model of a realtime vehicle position.
+ */
+public class RealtimeVehiclePosition {
+
+    public final String vehicleId;
+    public final String label;
+    public final double lat;
+    public final double lon;
+
+    /**
+     * Speed in meters per second
+     */
+    public final Double speed;
+
+    /**
+     * Bearing, in degrees, clockwise from North, i.e., 0 is North and 90 is East. This can be the
+     * compass bearing, or the direction towards the next stop or intermediate location.
+     */
+    public final Double heading;
+
+    /**
+     * When the realtime position was recorded
+     */
+    public final Instant time;
+
+    /**
+     * Status of the vehicle, ie. if approaching the next stop or if it is there already.
+     */
+    public final StopStatus stopStatus;
+
+    public final StopLocation nextStop;
```

Comment:
Should we also store the reference to the trip + operating day here, so that we could link to those in the API?